papaya-web/src/main/java/pl/edu/agh/papaya/rest/sprints/service/SprintsRestService.java

```diff
@@ -159,47 +210,30 @@ private SprintSummaryDto createSprintSummaryDto(Sprint currentSprint,
                 .map(userAvailabilityMapper::emptyAvailability).collect(Collectors.toList());
     }
 
-    private Sprint getValidSprint(Long sprintId) {
-        return sprintService.getById(sprintId)
-                .orElseThrow(ResourceNotFoundException::new);
-    }
-
     public ResponseEntity<SprintDto> modifySprint(SprintDto sprintDto, Long projectId, Long sprintId) {
-        Project project = projectsRestService.getValidProject(projectId);
+        Sprint sprint = getValidSprint(projectId, sprintId);
 
-        if (!project.isAdmin(userContext.getUser())) {
+        if (!sprint.getProject().isAdmin(userContext.getUser())) {
             throw new ForbiddenAccessException();
         }
 
         if (sprintDto.getSprintState() == SprintStateDto.CLOSED) {
-            Duration timeBurned = Duration.ofMinutes(
-                    AssertionUtil.requireNonNegative("timeBurned", sprintDto.getTimeBurned())
-            );
-
-            Duration finalTimePlanned = Duration.ofMinutes(
-                    AssertionUtil.requireNonNegative("finalTimePlanned", sprintDto.getFinalTimePlanned())
-            );
-
-            return closeSprint(sprintId, timeBurned, finalTimePlanned);
+            return closeSprint(sprint, sprintDto.getTimeBurned(), sprintDto.getEstimatedTimePlanned(),
+                    sprintDto.getFinalTimePlanned());
         }
 
-        throw new BadRequestException("Can not modify unclosed sprint");
+        throw new BadRequestException("The only available sprint modification is closing the sprint");
     }
 
-    private ResponseEntity<SprintDto> closeSprint(Long sprintId, Duration timeBurned, Duration finalTimePlanned) {
-        Sprint sprint = getValidSprint(sprintId);
-
-        if (sprint.getDateClosed() != null) {
-            throw new IllegalStateException("Sprint already closed");
-        }
-
+    private ResponseEntity<SprintDto> closeSprint(Sprint sprint, Long timeBurned, Long estimatedTimePlanned,
+            Long finalTimePlanned) {
         LocalDateTime timeNow = LocalDateTime.now();
-
-        if (sprint.getSprintState(timeNow) != SprintState.FINISHED) {
-            throw new IllegalStateException("Sprint cannot be closed before it has ended");
+        try {
+            Sprint changedSprint =
+                    sprintService.closeSprint(sprint, timeBurned, estimatedTimePlanned, finalTimePlanned, timeNow);
+            return ResponseEntity.ok(sprintMapper.mapToApi(changedSprint));
+        } catch (IllegalStateException e) {
```

Comment:
Tu to `IllegalStateException` leciało na front i było rzucane w `SprintsRestService`, a wydaje mi się, że w `SprintService` ładniej, więc przeniosłam te wyjątki do środeczka `SprintService`, a tu leci coś z losowym kodem HTTP, który był pod ręką akurat